### PR TITLE
Update dependencies, add Timber logging

### PR DESCRIPTION
**Pull request by joshsmith**
# What's in this PR?

- Updates a bunch of dependencies
- Update Elixir, Erlang/OTP versions
- Fix tests to work with new Poison
- Install new Timber version
- Add user context for Timber

Poison needs to be overridden to work with Sentry.

### CODE REVIEW

File: config/dev.exs [hunk 1]
```diff
@@ -67,7 +67,7 @@ config :code_corps,
 # If the dev environment has no CLOUDEX_API_KEY set, we want the app
 # to still run, with cloudex in test API mode
 if System.get_env("CLOUDEX_API_KEY") == nil do
-  IO.puts("NOTE: No Cloudex configuration found. Cloudex is runnning in test mode.")
+  IO.puts("NOTE: No Cloudex configuration found. Cloudex is running in test mode.")
```
Comment (begedin) [inline config/dev.exs:70]: 🥇

File: lib/code_corps_web/guardian_serializer.ex [hunk 2]
```diff
@@ -10,6 +10,24 @@ defmodule CodeCorpsWeb.GuardianSerializer do
...
+
+    if user do
+      name = full_name(user)
+      %Timber.Contexts.UserContext{id: user.id, email: user.email, name: name}
+      |> Timber.add_context()
+    end
+
+    {:ok, user}
+  end
   def from_token(_), do: {:error, "Unknown resource type"}
```
Comment (begedin) [inline lib/code_corps_web/guardian_serializer.ex:24]: This is sort of adding a side-effect to the serializer.

I would be more comfortable specifying a separate, more explicit plug which adds this context.

Something like the combination of `CodeCorpsWeb.Plug.Segment` and `CodeCorps.Analytics.SegmentPlugTracker`, except in a single module.

**Review by begedin — DISMISSED** [review]
I like the addition, but I also have a couple of suggestions about making it more explicit.

File: lib/code_corps_web/guardian_serializer.ex (hunk 2 above)
Comment (joshsmith) [reply lib/code_corps_web/guardian_serializer.ex:24]: In the interest of getting this merged, could we open an issue to circle back and address it? This, for me, is lower priority than other things on the totem pole.

Comment (begedin) [reply lib/code_corps_web/guardian_serializer.ex:24]: #932

**Review by begedin — DISMISSED** [review]

**Comment by begedin** [comment]
Created issue. @JoshSmith Feel free to merge when ready.

**Pull request merged**